gffquant/db/annotation_db.py

```
@@ -212,6 +254,29 @@ def get_features(self, category_id):
     def query_category(self, category_id):
         return self.db.categories.get(int(category_id))
 
+    def get_categories(self):
+        yield from self.db.categories.values()
+
+    def get_features(self, category=None):
+        for feat in self.db.features.values():
+            if category is None or feat.category_id == category:
+                yield feat
+
     @lru_cache(maxsize=10000)
```

Ruff (0.8.0)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)